pkg/util/lock.go

```diff
@@ -202,7 +200,7 @@ func (l *PgAdvisoryLock) getConn(connStr string, cur, maxRetries int) (*pgx.Conn
 		return nil, fmt.Errorf("max attempts reached. giving up on getting a db connection")
 	}
 
-	ctx, cancel := context.WithTimeout(context.Background(), waitForConnectionTimeout)
+	ctx, cancel := context.WithTimeout(context.Background(), l.connTimeout)
```

Comment:
I think the code should only impose a timeout if one isn't defined by the connection string. Remember a connection string can be define by the -db-* parameters (including -db-connection-timeout) OR by the -db-uri 
parameter. We should only impose a timeout here if the connection string doesn't have one. We can use something like:

```
      ctx := context.Background()
      cfg, err := pgx.ParseConfig(connStr)
      if cfg.ConnectTimeout == 0 {
            ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
            defer cancel()
      }

```